### PR TITLE
feat(frontend): Merge `LogicalUnion` all with `LogicalValues` inputs into `LogicalValues` 

### PR DESCRIPTION
**Pull request by jon-chuang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
Merge `LogicalUnion` all with `LogicalValues` inputs into `LogicalValues` 

Depends on: https://github.com/risingwavelabs/risingwave/pull/7912

## Checklist

- [X] I have written necessary rustdoc comments
- [X] I have added necessary unit tests and integration tests
- [X] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features).
- [X] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [X] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [X] My PR **DOES NOT** contain user-facing changes.

## Fixes
https://github.com/risingwavelabs/risingwave/issues/7859

### CODE REVIEW

File: src/frontend/src/optimizer/mod.rs [hunk 1]
```diff
@@ -410,6 +412,8 @@ impl PlanRoot {
                 OverAggToTopNRule::create(),
                 ProjectMergeRule::create(),
                 ProjectEliminateRule::create(),
+                TrivialProjectToValuesRule::create(),
```
Comment (jon-chuang) [inline src/frontend/src/optimizer/mod.rs:415]: not sure if we need to put it here. I guess there is no harm.

File: src/frontend/src/optimizer/rule/union_input_values_merge_rule.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            && union.all()
+        {
+            let mut rows = vec![];
+            union.inputs().iter().for_each(|v| {
+                rows.extend_from_slice(v.as_logical_values().unwrap().rows());
+            });
+            Some(LogicalValues::new(rows, union.schema().clone(), union.ctx()).into())
+        } else {
+            None
+        }
```
Comment (xxchan) [inline src/frontend/src/optimizer/rule/union_input_values_merge_rule.rs:37]: What about
```suggestion
        if !union.all() {
            return None;
        }

        let mut rows = vec![];
        for v in union.inputs() {
            rows.extend_from_slice(v.as_logical_values()?.rows());
        }
        Some(LogicalValues::new(rows, union.schema().clone(), union.ctx()).into())
    
```

File: src/frontend/planner_test/tests/testdata/union.yaml [hunk 3]
```diff
@@ -206,70 +206,22 @@
...
-      | └─LogicalValues { rows: [[]], schema: Schema { fields: [] } }
-      ├─LogicalProject { exprs: [2:Int32] }
-      | └─LogicalValues { rows: [[]], schema: Schema { fields: [] } }
-      ├─LogicalProject { exprs: [3:Int32] }
-      | └─LogicalValues { rows: [[]], schema: Schema { fields: [] } }
-      ├─LogicalProject { exprs: [4:Int32] }
-      | └─LogicalValues { rows: [[]], schema: Schema { fields: [] } }
-      └─LogicalProject { exprs: [5:Int32] }
-        └─LogicalValues { rows: [[]], schema: Schema { fields: [] } }
+    └─LogicalValues { rows: [[1:Int32], [2:Int32], [3:Int32], [4:Int32], [5:Int32], [5:Int32]], schema: Schema { fields: [1:Int32:Int32] } }
```
Comment (xxchan) [inline src/frontend/planner_test/tests/testdata/union.yaml:222]: The schema: `fields: [1:Int32:Int32]` looks wierd 🤔

**Review by xxchan — APPROVED** [review]

File: src/frontend/planner_test/tests/testdata/union.yaml (hunk 3 above)
Comment (jon-chuang) [reply src/frontend/planner_test/tests/testdata/union.yaml:222]: Yes, it's due to project. I decided not to change. Discussed with tianshuo on this.

**Review by st1page — APPROVED** [review]
LGTM

File: src/frontend/src/optimizer/rule/union_input_values_merge_rule.rs (hunk 2 above)
Comment (jon-chuang) [reply src/frontend/src/optimizer/rule/union_input_values_merge_rule.rs:37]: Thanks, good catch.

**Pull request merged**